lib/main.dart

```diff
@@ -22,9 +27,40 @@ void main({String env = 'prod'}) async {
   } catch (err) {
     print('$err');
   }
+
+  await Workmanager().initialize(
+    callbackDispatcher,
+    isInDebugMode: false
+  );
+
+  await Workmanager().registerPeriodicTask('backgroundTracking', 'backgroundTracking', frequency: Duration(hours: 4, minutes: 48));
+
   runApp(MyApp());
 }
 
+void callbackDispatcher() {
+  Workmanager().executeTask((task, inputData) async {
+    await Firebase.initializeApp();
+
+    switch (task) {
+      case 'backgroundTracking':
+        var permission = await Geolocator.checkPermission();
+        var isBgTrackingEnabled = await UserManager.getTracking();
```

Comment:
Could you double check that `UserManager.getTracking()` return True when tracking is enabled? I'm getting confused in the `setTracking` method, which seems to set the variable `trackingDisabled`... Not sure

Reply:
Ups... nevermind, I hadn't seen that you modified the UserManager when i posted the comment...